api/src/main/java/jakarta/data/BasicRestrictionRecord.java

```diff
@@ -36,4 +36,33 @@ record BasicRestrictionRecord<T>(
     BasicRestrictionRecord(String field, Operator comparison, Object value) {
         this(field, false, comparison, value);
     }
+
+    @Override
+    public Restriction<T> negate() {
+        boolean newNegation = isNegated;
+        Operator newComparison;
```

Comment:
This is incredibly anal, I know, but can we avoid the use of non-`final` instance vars here, please?

Also: remind we what version of Java we're targeting? Can we still not use `switch` expressions? I thought we could now.

Reply:
> This is incredibly anal, I know, but can we avoid the use of non-`final` instance vars here, please?

The non-final variable went away after I moved the negation detail to Operator, so this is covered now.

> Also: remind we what version of Java we're targeting? Can we still not use `switch` expressions? I thought we could now.

Jakarta EE 11 targets Java SE 21 and Java SE 17.
Jakarta EE 12 will add Java SE 25 (and I assume drop 17 but not really sure).

It would make sense for Data 1.1 to aim for compatibility with all 3 of these Java levels.  I don't know of any concerns with `switch` statements for these levels.

Reply:
OK, so we can use `switch` expressions then, right?

Reply:
Yes, I accepted your suggestion to use the more concise `->` instead of `:` and followed that up with a commit to add `;` at the end which was also needed.  This is also nice because allowed us to get past the code checking police without needing to add the useless default throw exception.  It is actually better without the default because if someone adds a new enumerated value to Operator at some point, this method will not compile until they also add the implementation for negating it.